File: 240410/원점으로부터의 거리/distance-from-origin.py
N = int(input())
points = [list(map(int, input().split())) + [num] for num in range(1, N + 1)]

points.sort(key = lambda x: int(abs(x[0])) + int(abs(x[1])))

for x, y, num in points:
    print(num)

File: 260105/원점으로부터의 거리/distance-from-origin.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct Points
{
    int x;
    int y;
    int num;
};

int main() 
{
    int N;
    cin >> N;
    
    vector<Points> points;
    points.reserve(N);

    // Please write your code here.
    for(int i = 1; i <= N; i++)
    {
        int x, y;
        cin >> x >> y;
        points.push_back({x, y, i});
    }

    sort(points.begin(), points.end(), 
        [](const Points& a, const Points& b)
            {
                return abs(a.x) + abs(a.y) < abs(b.x) + abs(b.y);
            });

    for(const auto& p: points)
    {
        cout << p.num << endl;
    }
    return 0;
}
